samlsp/samlsp.go

```diff
@@ -101,9 +101,9 @@ func New(opts Options) (*Middleware, error) {
 			}
 
 			err = fmt.Errorf("no entity found with IDPSSODescriptor")
-			for _, e := range entities.EntityDescriptors {
+			for i, e := range entities.EntityDescriptors {
 				if len(e.IDPSSODescriptors) > 0 {
-					entity = &e
```

Comment:
Yeah, this is a bug here for sure. This is taking the address of the locally scoped variable e.

https://developmentality.wordpress.com/2014/03/03/go-gotcha-1-variable-shadowing-within-inner-scope-due-to-use-of-operator/